x-pack/plugins/ml/public/application/datavisualizer/stats_table/hooks/use_data_viz_chart_theme.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import darkTheme from '@elastic/eui/dist/eui_theme_dark.json';
+import lightTheme from '@elastic/eui/dist/eui_theme_light.json';
+import type { PartialTheme } from '@elastic/charts';
+import { useUiSettings } from '../../../contexts/kibana';
+export const useDataVizChartTheme = (): PartialTheme => {
+  const isDarkTheme = useUiSettings().get('theme:darkMode');
```

Comment:
To be using `useUiSettings` in a hook like this, shouldn't it be watching for changes in the settings?
Like this function:
https://github.com/elastic/kibana/blob/f7fdda5db7e41b04f0f37cc180b441e8ed0c9321/x-pack/plugins/ml/public/application/components/color_range_legend/use_color_range.ts#L192

Otherwise it doesn't need to be in a hook.

Reply:
You're right. This is now updated here https://github.com/elastic/kibana/pull/87598/commits/e09dc6ff939c54663d632f74d76e4f2caea4a9eb